src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -50,4 +54,12 @@ public BookDto updateById(Long id, CreateBookRequestDto bookRequestDto) {
     public void deleteById(Long id) {
         bookRepository.deleteById(id);
     }
+
+    @Override
+    public List<BookDto> searchByParameters(BookSearchParametersDto params) {
+        Specification<Book> bookSpecification = bookSpecificationBuilder.build(params);
+        return bookRepository.findAll(bookSpecification).stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
```

Comment:
let's teach our mapper to map the entire list instead of iterating through the list manually and calling single-map operation